capstone/client/src/components/ListPage.js

```diff
@@ -74,6 +76,17 @@ class ListPage extends Component {
         alert: (<Alert severity="error">Please select at least one item!</Alert>),
       })
     } else {
+      const response = axios.post(
+        '/api/v1/create-or-update-user-list-servlet',
+        { 
+          userId: 1,
+          userList: {
+            listId: 1,
```

Comment:
When the user first creates a list, how is the UI going to know what's the listId? Or how can it create a listId so it's unique? it's often better to assign the listId generation to the backend, so in your servlet (or database utils) you can have some helper function that generates a unique ID, and then maybe return the ID to the UI for subsequent operations. One possible unique ID is the userId + timestamp, for example. Or you can ask the DB to generate a unique key for you.

Reply:
(maybe timestamp is not great because it's always increasing and will generate hotspots)
https://cloud.google.com/spanner/docs/schema-and-data-model#choosing_a_primary_key